baby-gru/src/WebGLgComponents/mgWebGL.tsx

```diff
@@ -3554,6 +3554,34 @@ export class MGWebGL extends React.Component implements webGL.MGWebGL {
         if (drawScene) this.drawScene();
     }
 
+    drawZoomFrame(oldZoom: number, newZoom: number, iframe: number) {
+        const deltaZoom = (newZoom - oldZoom) / this.nAnimationFrames
+        const currentZoom = oldZoom + deltaZoom * iframe
+        this.zoom = currentZoom
+        this.drawScene()
+        if (iframe < this.nAnimationFrames) {
+            const fieldDepthFront = 8
+            const fieldDepthBack = 21
+            this.set_fog_range(this.fogClipOffset - (this.zoom * fieldDepthFront), this.fogClipOffset + (this.zoom * fieldDepthBack))
+            this.set_clip_range(0 - (this.zoom * fieldDepthFront), 0 + (this.zoom * fieldDepthBack))
+            requestAnimationFrame(this.drawZoomFrame.bind(this, oldZoom, newZoom, iframe + 1))
+        } else {
+            const zoomChanged = new CustomEvent("zoomChanged", {
+                "detail": {
+                    oldZoom,
+                    newZoom
+                }
+            });
+            document.dispatchEvent(zoomChanged);    
+        }
+    }
+
+    setZoomAnimated(newZoom: number) {
+        const oldZoom = this.zoom
+        this.nAnimationFrames = 15
+        requestAnimationFrame(this.drawZoomFrame.bind(this, oldZoom, newZoom, 1))
+    }
```

Comment:
@stuartjamesmcnicholas can you check that this is the best way of doing things? The way I intend to use it is together with `setOriginAnimated` like this:
```javascript
if (animate) {
   if (setZoom) this.glRef.current.setZoomAnimated(0.4)
   this.glRef.current.setOriginAnimated(selectionCentre);
}
```
I'm not sure if this will result in unnecessary calls to `drawScene` ?

Reply:
Would it be better to have a `setOriginAndZoomAnimated` where both the zoom and the origin are changed at the same time and then `drawScene()` is only called once?

Reply:
Finally I've refactored this into a two separate methods `setZoomAnimated` and `setOriginAndZoomAnimated`